File: Daily Challanges/2026-07-July/2026-07-17.py
from datetime import date


def days_until_birthday(today, birthday):
    year, month, day = map(int, today.split("-"))
    birth_month, birth_day = map(int, birthday.split("/"))
    today_date = date(year, month, day)

    def next_birthday(start_year):
        y = start_year
        while True:
            try:
                return date(y, birth_month, birth_day)
            except ValueError:
                y += 1

    birthday_date = next_birthday(year)

    if birthday_date <= today_date:
        birthday_date = next_birthday(birthday_date.year + 1)

    return (birthday_date - today_date).days


print(days_until_birthday("2026-07-16", "9/7"))
print(days_until_birthday("2026-07-16", "3/22"))
print(days_until_birthday("2026-07-16", "7/16"))
print(days_until_birthday("2024-02-28", "3/1"))
print(days_until_birthday("2023-04-24", "12/30"))
print(days_until_birthday("2024-03-01", "2/29"))
print(days_until_birthday("2096-03-01", "2/29"))

File: Daily Challanges/2026-07-July/2026-07-17.js
function daysUntilBirthday(today, birthday) {
  const [year, month, day] = today.split("-").map(Number);
  const [birthMonth, birthDay] = birthday.split("/").map(Number);
  const todayDate = new Date(year, month - 1, day);

  function nextBirthday(startYear) {
    let y = startYear;

    while (true) {
      const d = new Date(y, birthMonth - 1, birthDay);
      if (
        d.getFullYear() === y &&
        d.getMonth() === birthMonth - 1 &&
        d.getDate() === birthDay
      ) {
        return d;
      }
      y++;
    }
  }

  let birthdayDate = nextBirthday(year);

  if (birthdayDate <= todayDate) {
    birthdayDate = nextBirthday(birthdayDate.getFullYear() + 1);
  }

  return Math.round((birthdayDate - todayDate) / 86400000);
}

console.log(daysUntilBirthday("2026-07-16", "9/7"));
console.log(daysUntilBirthday("2026-07-16", "3/22"));
console.log(daysUntilBirthday("2026-07-16", "7/16"));
console.log(daysUntilBirthday("2024-02-28", "3/1"));
console.log(daysUntilBirthday("2023-04-24", "12/30"));
console.log(daysUntilBirthday("2024-03-01", "2/29"));
console.log(daysUntilBirthday("2096-03-01", "2/29"));
